# Sales Forecasting for Retail Decision-Making  
## A Business Intelligence Case Study Using Walmart Sales Data  

**Module:** Business Intelligence and Data Mining (UFCFMM)  
**Assessment:** Group Project Portfolio  
**Team Members:** Dan Shipton, Samuel Fitzgerlad, Nifemi Omoshuli  
**Dataset Source:** Kaggle – Walmart Sales Dataset  

---

## 1. Executive Summary

## 2. Business Context & Problem Definition


The Business context 

Walmart is a large-scale retail organisation that operates within a highly competitive and low-margin market where operational efficiency and demand responsiveness is key to lasting profitability. Sales forecasting is a strategic asset that underpins inventory management, workforce allocation, supply chain coordination, promotional planning and financial forecasting.  



An inaccurate financial forecast can cause substantial operational inefficiencies. An overwhelming amount of demands will lead to overstocking, an increased holding cost, markdown losses and waste products. Also, undermining the forecast can lead to stockouts, lost revenue and loss of customer satisfaction. In short, even small forecasting errors can turn into a significant financial impact especialy when aggregated across department across department and regions.  



The retail demand is influenced by multiple factors. 
Seasonal change (Holiday periods)  
Unemployment inflation. 



External costs (Fuel prices. The higher the cost of fuel, the less amount the customer will spend.)  


the business problem 

This project will address the following problem. How can historical weekly sales data across Walmart stores be analysed for firstly the forecast future sales with acceptable predictive accuracy? Secondly, identify the key structure and the economic factors that would influence the sales variability, therefore enabling data-driven operational planning  

The problem is divided into two different sections.  
Predictive dimensions (Generating the forward-looking sales forecast)  
Diagnostic dimensions (Understand what drives sales variation)  

By combining the forecasting with explanatory analysis, we move beyond technical modelling and towards actionable business intelligence, aligning directly with the learning outcomes of investigating real-world applications of data mining in business contexts.

Local store Intelligence  

A competent business intelligence system will be able to identify the demand drivers and will be able to produce relable forecast and provide a competitive advantage.


## 3. Dataset Description & Justification

In [2]:
## 4. Technical Setup & Data Loading

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Load dataset
df = pd.read_csv("Walmart.csv")

df.head()

,Store,Date,Weekly_Sales,Holiday_Flag,Temperature,Fuel_Price,CPI,Unemployment
0,1,05-02-2010,1643690.90,0,42.31,2.572,211.096358,8.106
1,1,12-02-2010,1641957.44,1,38.51,2.548,211.242170,8.106
2,1,19-02-2010,1611968.17,0,39.93,2.514,211.289143,8.106
3,1,26-02-2010,1409727.59,0,46.63,2.561,211.319643,8.106
4,1,05-03-2010,1554806.68,0,46.50,2.625,211.350143,8.106


## 5. Data Quality Assessment


In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 6435 entries, 0 to 6434
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Store         6435 non-null   int64  
 1   Date          6435 non-null   str    
 2   Weekly_Sales  6435 non-null   float64
 3   Holiday_Flag  6435 non-null   int64  
 4   Temperature   6435 non-null   float64
 5   Fuel_Price    6435 non-null   float64
 6   CPI           6435 non-null   float64
 7   Unemployment  6435 non-null   float64
dtypes: float64(5), int64(2), str(1)
memory usage: 402.3 KB


In [4]:
df.isnull().sum()

Store           0
Date            0
Weekly_Sales    0
Holiday_Flag    0
Temperature     0
Fuel_Price      0
CPI             0
Unemployment    0
dtype: int64

In [5]:
df.describe()

,Store,Weekly_Sales,Holiday_Flag,Temperature,Fuel_Price,CPI,Unemployment
count,6435.000000,6.435000e+03,6435.000000,6435.000000,6435.000000,6435.000000,6435.000000
mean,23.000000,1.046965e+06,0.069930,60.663782,3.358607,171.578394,7.999151
std,12.988182,5.643666e+05,0.255049,18.444933,0.459020,39.356712,1.875885
min,1.000000,2.099862e+05,0.000000,-2.060000,2.472000,126.064000,3.879000
25%,12.000000,5.533501e+05,0.000000,47.460000,2.933000,131.735000,6.891000
50%,23.000000,9.607460e+05,0.000000,62.670000,3.445000,182.616521,7.874000
75%,34.000000,1.420159e+06,0.000000,74.940000,3.735000,212.743293,8.622000
max,45.000000,3.818686e+06,1.000000,100.140000,4.468000,227.232807,14.313000


### Data Quality Summary

The dataset contains 6,435 observations across 45 Walmart stores, with 8 variables describing weekly sales and related economic indicators.

No missing values were detected across any variables, indicating a complete dataset suitable for modelling without imputation.

Weekly sales range from approximately £209,986 to £3,818,686, with an average weekly sales value of approximately £1,046,965. This substantial variability highlights the operational importance of accurate demand forecasting in large retail environments.

All economic indicators (Temperature, Fuel Price, CPI, and Unemployment) appear numerically consistent and require no immediate cleaning. However, the Date variable was initially stored as a string and has been converted to datetime format to enable time series analysis.


In [6]:
df.info()


<class 'pandas.DataFrame'>
RangeIndex: 6435 entries, 0 to 6434
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Store         6435 non-null   int64  
 1   Date          6435 non-null   str    
 2   Weekly_Sales  6435 non-null   float64
 3   Holiday_Flag  6435 non-null   int64  
 4   Temperature   6435 non-null   float64
 5   Fuel_Price    6435 non-null   float64
 6   CPI           6435 non-null   float64
 7   Unemployment  6435 non-null   float64
dtypes: float64(5), int64(2), str(1)
memory usage: 402.3 KB


In [7]:
df.isnull().sum()


Store           0
Date            0
Weekly_Sales    0
Holiday_Flag    0
Temperature     0
Fuel_Price      0
CPI             0
Unemployment    0
dtype: int64

In [8]:
df.describe()


,Store,Weekly_Sales,Holiday_Flag,Temperature,Fuel_Price,CPI,Unemployment
count,6435.000000,6.435000e+03,6435.000000,6435.000000,6435.000000,6435.000000,6435.000000
mean,23.000000,1.046965e+06,0.069930,60.663782,3.358607,171.578394,7.999151
std,12.988182,5.643666e+05,0.255049,18.444933,0.459020,39.356712,1.875885
min,1.000000,2.099862e+05,0.000000,-2.060000,2.472000,126.064000,3.879000
25%,12.000000,5.533501e+05,0.000000,47.460000,2.933000,131.735000,6.891000
50%,23.000000,9.607460e+05,0.000000,62.670000,3.445000,182.616521,7.874000
75%,34.000000,1.420159e+06,0.000000,74.940000,3.735000,212.743293,8.622000
max,45.000000,3.818686e+06,1.000000,100.140000,4.468000,227.232807,14.313000


## 6. Exploratory Data Analysis (EDA)


## 7. Feature Engineering & Preprocessing


## 8. Baseline Forecast Model


## 9. Forecasting Models


## 10. Model Evaluation (Technical & Business)


## 11. Strategic Recommendations


## 12. Ethical, Privacy & Security Evaluation


## 13. Conclusion


## 14. References
